### PR TITLE
dbex/78112 none location 1990

### LINKED ISSUE #78112: Toxic Exposure: 'Add None of these locations' to gulf war 1990 locations page
## Issue Description
- Add the 'None of these locations' checkbox to the `gulf-war-hazard-1990` page
- If Veteran selects one or more locations and also checks the none checkbox, display error message in mock
- If no checkboxes are selected OR if only none is selected, subsequent pages (details pages and summary page) should not display

## Front end tasks
- Ensure the page is styled according to design spec. 
- Ensure the back end data from API is displayed in local and review instance
- Ensure unit tests are available. 
- Ensure your code changes are covered by E2E tests
- Run axe checks using the Chrome or Firefox browser plugin
- Test for color contrast and color blindness issues
- Zoom layouts to 400% at 1280px width
- Test with 1 or 2 screen readers
- Navigate using the keyboard only

## Acceptance Criteria
- [ ] Designer comments on this issue to approve page styling 
- [ ] Unit tests pass
- [ ] E2E tests cover current code and regression
- [ ] End-to-end tests show 0 violations.
- [ ] The data returned matches API response (for given user or scenario)
- [ ] All axe checks pass
- [ ] All color contrast checks pass
- [ ] All zoom testing passes
- [ ] All keyboard checks pass 
- [ ] All screenreader checks pass

## Appendix
<details> 

 <summary>Ensure your code changes are covered by E2E tests (expand)</summary>
 
 - Add E2E tests if none exist for this addition/change. 
 
 - Update existing E2E tests if this code will change functionality. 
 
 - Include axe checks, including hidden content
 
</details>

<details> 

 <summary> Run axe checks using the Chrome or Firefox browser plugin (expand)</summary>
 
- Ensure no heading levels are skipped.

- Ensure all buttons and labeled inputs use semantic HTML elements.

- Ensure all buttons, labeled elements and images are identified using HTML semantic markers or ARIA roles.

- Ensure form fields have clearly defined boundaries or outlines.

- Ensure form fields do not use placeholder text. 

- Ensure form fields have highly visible and specific error states. 

</details>

<details> 

 <summary> Test for color contrast and color blindness issues (expand) </summary>
 - All text has appropriate contrast. 

</details>

<details> 

 <summary> Zoom layouts to 400% at 1280px width (expand)</summary> 
 - Ensure readability and usability are supported when zoomed up to 400% at 1280px browser width
 - Ensure no content gets focused offscreen or is hidden from view.
 
</details>

<details> 

 <summary> Test with 1 or 2 screen readers (expand)</summary> 
 
 - Ensure the page includes a skip navigation link. 
 
 - Ensure all links are properly descriptive. 
 
 - Ensure screen reader users can hear the text equivalent for each image conveying information. 

 - Ensure screen reader users can hear the text equivalent for each image button. 

 - Ensure screen reader users can hear labels and instructions for inputs. 
 
 - Ensure purely decorative images are not announced by the screenreader. 
 
</details>

<details> 

 <summary>Navigate using the keyboard only (expand)</summary> 
 
 - Ensure all links (navigation, text and/or image), form controls and page functions can be reached with the tab key in a logical order. 
 
 - Ensure all links (navigation, text and/or image), form controls and page functions can be triggered with the spacebar, enter key, or arrow keys.
 
 - Ensure all interactive elements can be reached with the tab key in a logical order 
 
 - Ensure all interactive elements can be triggered with the spacebar, enter key, or arrow keys.
 
 - Ensure focus is always visible and appears in logical order.
 
 - Ensure each interactive element has visible focus state which appears in logical order.
 

</details>

---
## How to configure this issue
- [ ] **Attached to a Milestone** (when will this be completed?)
- [ ] **Attached to an Epic** (what body of work is this a part of?)
- [ ] **Labeled with Team** (`product support`, `analytics-insights`, `operations`, `service-design`, `Console-Services`, `tools-fe`)
- [ ] **Labeled with Practice Area** (`backend`, `frontend`, `devops`, `design`, `research`, `product`, `ia`, `qa`, `analytics`, `contact center`, `research`, `accessibility`, `content`)
- [ ] **Labeled with Type** (`bug`, `request`, `discovery`, `documentation`, etc.)

### PR DESCRIPTION
**Pull request by michael-wojcik**
- **📦 NEW: add none location to gulf war 1990 options**
- **📦 NEW: add validateGulfWar1990Locations**
- **👌 IMPROVE: hide gulfWar1990 location, summary pages if `none` selected**
- **📦 NEW: add unit test for gulfWar1990 `none` and other location validation**
- **📦 NEW: add unit tests for hiding gulfWar1990 detail, summary pages when `none` location selected**

## Summary

- Add "None of these locations" to list of Gulf War 1990 locations for Toxic Exposure portion of 526EZ form
- DBEX TREX
- Flipper: `disability_526_toxic_exposure`

## Related issue(s)

- department-of-veterans-affairs/va.gov-team#78112

## Testing done

- If "None of these locations" is selected, pressing "Continue" button should result in proceeding to next section without seeing additional pages related to the Gulf War 1990 location.
- If "None of these locations" is selected along with one or more other locations, pressing "Continue" button should result in an error message.

## Screenshots

_Note: This field is mandatory for UI changes (non-component work should NOT have screenshots)._

|         | Before | After |
| ------- | ------ | ----- |
| Mobile  | Screenshot 2024-05-09 at 11 48 04 | Screenshot 2024-05-09 at 11 54 19 |
| Desktop | Screenshot 2024-05-09 at 11 47 32 | Screenshot 2024-05-09 at 11 53 55 |

## What areas of the site does it impact?

526EZ

### Quality Assurance & Testing

- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x] Linting warnings have been addressed
- [x] Documentation has been updated ([link to documentation](#) \*if necessary)
- [x] Screenshot of the developed feature is added
- [x] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [x] Browser console contains no warnings or errors.
- [x] Events are being sent to the appropriate logging solution
- [x] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [x] Did you login to a local build and verify all authenticated routes work as expected with a test user

### CODE REVIEW

File: src/applications/disability-benefits/all-claims/content/toxicExposure.jsx [hunk 1]
```diff
@@ -239,6 +242,25 @@ export function validateTEConditions(errors, formData) {
...
+export function validateGulfWar1990Locations(errors, formData) {
+  const { gulfWar1990 = {} } = formData;
+
+  if (
+    gulfWar1990.none === true &&
+    Object.values(gulfWar1990).filter(value => value === true).length > 1
+  ) {
+    errors.gulfWar1990.addError(noneAndLocationError);
+  }
+}
```
Comment (mchae-nava) [inline src/applications/disability-benefits/all-claims/content/toxicExposure.jsx:262]: Looks sound so far. Is the intention to keep this strictly as a frontend validation, or will we have to account for a None type being submitted as well?

File: src/applications/disability-benefits/all-claims/tests/pages/toxicExposure/gulfWar1990Locations.unit.spec.jsx [hunk 2]
```diff
@@ -86,4 +87,26 @@ describe('Gulf War 1990 Locations', () => {
...
+      target: {
+        checked: true,
+        dataset: { key: 'none' },
+      },
+      detail: { checked: true },
+    });
+
+    await userEvent.click(getByText('Submit'));
+    expect($('va-checkbox-group').error).to.equal(noneAndLocationError);
+  });
```
Comment (mchae-nava) [inline src/applications/disability-benefits/all-claims/tests/pages/toxicExposure/gulfWar1990Locations.unit.spec.jsx:111]: Appreciate the unit tests! Looks good

**Review by mchae-nava — COMMENTED** [review]
Have you already accounted for the accessibility tests in the ticket subtasks? Also want to double check that you made a pass to have designs match spec

File: src/applications/disability-benefits/all-claims/content/toxicExposure.jsx [hunk 3]
```diff
@@ -49,6 +49,9 @@ export const gulfWar2001Question =
 export const noneAndConditionError =
   'You selected a condition, and you also selected “I’m not claiming any conditions related to toxic exposure.” You’ll need to uncheck one of these options to continue.';
 
+export const noneAndLocationError =
+  'You selected a location, and you also selected "none of these locations." You’ll need to uncheck one of these options to continue.';
```
Comment (christinec-fftc) [inline src/applications/disability-benefits/all-claims/content/toxicExposure.jsx:53]: I know this matches the mocks, but I wonder if the `.` should go outside of the quotes and whether 'none' should be capitalized (both so they match the checkbox value). Double check with Kate/Mike.

File: src/applications/disability-benefits/all-claims/content/toxicExposure.jsx (hunk 1 above)
Comment (christinec-fftc) [reply src/applications/disability-benefits/all-claims/content/toxicExposure.jsx:262]: Good question. There will be some backend work needed as well. See notes [here](https://app.zenhub.com/workspaces/disability-experience-63dbdb0a401c4400119d3a44/issues/gh/department-of-veterans-affairs/va.gov-team/72816)

File: src/applications/disability-benefits/all-claims/tests/content/toxicExposure.unit.spec.jsx [hunk 4]
```diff
@@ -554,6 +554,49 @@ describe('toxicExposure', () => {
...
+              condition: 'anemia',
+            },
+          ],
+          toxicExposureConditions: {
+            anemia: true,
+          },
+        };
+
+        expect(showGulfWar1990LocationDatesPage(formData)).to.be.false;
+      });
```
Comment (christinec-fftc) [inline src/applications/disability-benefits/all-claims/tests/content/toxicExposure.unit.spec.jsx:577]: 👆

File: src/applications/disability-benefits/all-claims/content/toxicExposure.jsx (hunk 3 above)
Comment (christinec-fftc) [reply src/applications/disability-benefits/all-claims/content/toxicExposure.jsx:53]: @Kate-allbee @mengA6 Is it ok if we change this error validation message to line up with the checkbox value? 
image

Comment (christinec-fftc) [inline src/applications/disability-benefits/all-claims/content/toxicExposure.jsx:53]:
```suggestion
  'You selected a location, and you also selected "None of these locations". You’ll need to uncheck one of these options to continue.';
```

**Review by christinec-fftc — APPROVED** [review]

**Pull request merged**